services/contacts/js/app.js

```diff
@@ -152,26 +152,37 @@
   // down to the SW thread, then back up here for processing, then back down to
   // be sent to the client. Yay us!
 
-  var processSWRequest = function(channel, evt) {
+  var processSWRequest = function(aAcl, aChannel, aEvt) {
     // We can get:
     // * oncontactchange
     // * getAll
     // * methodName
-    var request = evt.data.remoteData;
+    var request = aEvt.data.remoteData;
     var requestOp = request.data.operation;
+    var targetURL = aEvt.data.targetURL;
+
+    // TODO: Add resource access constraint
+    // It should return true if resource access is forbidden,
+    // false if it's allowed
+    var forbidCall = function(constraints) {
+      return false;
+    };
+
```

Comment:
Yes, I know it, but I did it this way because we need to add the resource access control to all services and I wanted to leave the skeleton done.
Also, I think if we add a default behavior it should be the reverse of this (forbid by default, not allow by default).